### PR TITLE
A few small tweaks that improves performance.

### PR DESCRIPTION
**Pull request by chrisvest**
The variance is quite rough, but I'm seeing 5-10% throughput improvement with these changes.

### CODE REVIEW

**Comment by chrisvest** [comment]
**Note:** We have tentatively decided to hold this one until _after_ 3.0.0 GA.

File: community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/CompatibilityFor3_0.scala [hunk 1]
```diff
@@ -183,8 +183,10 @@ trait CompatibilityFor3_0 {
 
   class ExecutionPlanWrapper(inner: ExecutionPlan_v3_0) extends ExecutionPlan {
 
+    // Eagerly create a single, shared index monitor, to avoid locking in the kernel monitors on every query execution.
+    val searchMonitor = kernelMonitors.newMonitor(classOf[IndexSearchMonitor])
```
Comment (davidegrohmann) [inline community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/CompatibilityFor3_0.scala]: private?

File: community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v3_0/TransactionBoundQueryContext.scala [hunk 2]
```diff
@@ -64,8 +64,9 @@ final class TransactionBoundQueryContext(val transactionalContext: Transactional
 
   val nodeOps = new NodeOperations
   val relationshipOps = new RelationshipOperations
+  lazy val nodeManager = transactionalContext.graph.getDependencyResolver.resolveDependency(classOf[NodeManager])
 
-  override def entityAccessor = transactionalContext.graph.getDependencyResolver.resolveDependency(classOf[NodeManager])
+  override def entityAccessor = nodeManager
```
Comment (davidegrohmann) [inline community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v3_0/TransactionBoundQueryContext.scala]: why not `override lazy val entityAccessor = transactionalContext.graph.getDependencyResolver.resolveDependency(classOf[NodeManager])`?

Comment (chrisvest) [inline community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v3_0/TransactionBoundQueryContext.scala]: I have only limited knowledge of Scala. I'm happy to make this change.

**Comment by chrisvest** [comment]
Thanks for your comments, @davidegrohmann

**Pull request merged**